mass-deployer/pkg/mass-deployer/load.go

```diff
@@ -0,0 +1,173 @@
+package deployer
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"path/filepath"
+	"sync"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/rs/zerolog/log"
+	"github.com/sethvargo/go-retry"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/deployer"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/graphql"
+	"github.com/threefoldtech/tfgrid-sdk-go/grid-client/workloads"
+	"gopkg.in/yaml.v3"
+)
+
+func RunLoader(ctx context.Context, cfg Config, debug bool, output string) error {
+	log.Info().Msg("Loading deployments")
+
+	tfPluginClient, err := setup(cfg, debug)
+	if err != nil {
+		return err
+	}
+	var nodeGroups []string
+	for _, group := range cfg.NodeGroups {
+		nodeGroups = append(nodeGroups, group.Name)
+	}
+
+	outputBytes, err := loadNodeGroupsInfo(ctx, tfPluginClient, nodeGroups, cfg.MaxRetries, output)
+	if err != nil {
+		return err
+	}
+
+	fmt.Println(string(outputBytes))
+	return os.WriteFile(output, outputBytes, 0644)
+}
+
+func loadNodeGroupsInfo(ctx context.Context, tfPluginClient deployer.TFPluginClient, nodegroups []string, maxRetries uint64, output string) ([]byte, error) {
+	trial := 1
+	nodeGroupsInfo := map[string][]vmOutput{}
+	failedGroups := map[string]string{}
+
+	// load contracts with node group name
+	for _, nodeGroup := range nodegroups {
```

Comment:
each of the nodeGroup can be loaded concurrently, you can do concurrent queries against chain state.